features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableCompoundButtonMapper.kt

```diff
@@ -36,27 +40,95 @@ internal abstract class CheckableCompoundButtonMapper<T : CompoundButton>(
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             view.buttonDrawable?.let {
                 checkBoxHeight = it.intrinsicHeight.toLong()
             }
         }
-        // minus the padding
-        checkBoxHeight -= MIN_PADDING_IN_PX * 2
         checkBoxHeight = checkBoxHeight.densityNormalized(pixelsDensity)
         return GlobalBounds(
-            x = viewGlobalBounds.x + MIN_PADDING_IN_PX.densityNormalized(pixelsDensity),
+            x = viewGlobalBounds.x,
             y = viewGlobalBounds.y + (viewGlobalBounds.height - checkBoxHeight) / 2,
             width = checkBoxHeight,
             height = checkBoxHeight
         )
     }
 
+    override fun getCheckableDrawable(view: T): Drawable? {
+        val originCheckableDrawable = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // drawable from [CompoundButton] can not be retrieved according to the state,
+            // so here two hardcoded indexes are used to retrieve "checked" and "not checked" drawables.
+            val checkableDrawableIndex = if (view.isChecked) {
+                CHECK_BOX_CHECKED_DRAWABLE_INDEX
+            } else {
+                CHECK_BOX_NOT_CHECKED_DRAWABLE_INDEX
+            }
+            (view.buttonDrawable?.constantState as? DrawableContainer.DrawableContainerState)?.getChild(
+                checkableDrawableIndex
+            )
+        } else {
+            // view.buttonDrawable is not available below API 23, so reflection is used to retrieve it.
+            try {
+                mButtonDrawableField?.get(view) as? Drawable
+            } catch (e: IllegalAccessException) {
+                internalLogger.log(
+                    level = InternalLogger.Level.ERROR,
+                    targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                    messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE },
+                    throwable = e
+                )
+                null
+            } catch (e: IllegalArgumentException) {
+                internalLogger.log(
+                    level = InternalLogger.Level.ERROR,
+                    targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                    messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE },
+                    throwable = e
+                )
+                null
+            }
+        }
+        return originCheckableDrawable?.let { cloneCheckableDrawable(view, it) } ?: run {
+            internalLogger.log(
+                level = InternalLogger.Level.ERROR,
+                targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE }
+            )
+            null
+        }
+    }
+
+    private fun cloneCheckableDrawable(view: T, drawable: Drawable): Drawable? {
+        return drawable.constantState?.newDrawable(view.resources)?.apply {
```

Comment:
is `newDrawable` expensive call CPU and memory-wise?

Reply:
Although different `Drawable` has different implementation of this function, most of them (including `AnimatedStateListDrawable` that `CompoundButton` is using) are just Instantiate a new instance of drawable, so I would say there is a memory allocation, but not expensive on CPU.
```
@Override
public Drawable newDrawable() {
    return new AnimatedStateListDrawableCompat(this, null);
}
```